app/models/vacation.rb

```diff
@@ -42,8 +42,18 @@ def deny!(user)
     end
   end
 
+  def cancelable?
+    self.pending? || approved_within_cancel_range?
+  end
+
   private
 
+  MINIMUM_DAYS_TO_CANCEL = 7
+
+  def approved_within_cancel_range?
+    (self.start_date.days_ago(MINIMUM_DAYS_TO_CANCEL) >= Date.today)
```

Comment:
why the parenthesis and you also don't need self here

Reply:
Just to let you know, you can also do `start_date.days_ago(MINIMUM_DAYS_TO_CANCEL).future?`